File: Javascript/0101-0150/118. Pascal's Triangle.js
// https://leetcode.com/problems/pascals-triangle/description/

var generate = function(numRows) {
    let result = [];
    const dummy = [];
    dummy.push(1);
    result.push(dummy);

    for(let row = 0; row < numRows-1; row++){
        const current = [];
        current.push(1);

        for(let index=0; index<result.length-1; index++){
            current.push(result[row][index]+result[row][index+1]);
        }
        current.push(1);
        result.push(current);
    }

    return result;
};

File: Python/0101-0150/118. Pascal's Triangle.py
# https://leetcode.com/problems/pascals-triangle/description/

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = []
        dummy = []
        dummy.append(1)
        result.append(dummy)

        for row in range(0, numRows-1):
            current = []
            current.append(1)

            for index in range(0, len(result)-1):
                current.append(result[row][index]+result[row][index+1])
            
            current.append(1)
            result.append(current)
        
        return result
   
